devito/ir/support/space.py

```diff
@@ -492,7 +492,12 @@ def project(self, cond):
         else:
             func = lambda i: i in cond
         intervals = [i for i in self.intervals if func(i.dim)]
-        sub_iterators = {k: v for k, v in self.sub_iterators.items() if func(k)}
+        sub_iterators = {}
+        for k, v in self.sub_iterators.items():
+            if func(k) and len(v) == 0:
+                sub_iterators.update({k: v})
+            elif func(k) and len(v) > 0:
+                sub_iterators.update({k: [] if v[0].dim.is_Conditional else v})
```

Comment:
@FabioLuporini this seems to fix the problem for conditionals and modulo. Not sure if it is the right place to do it though.